test/models/user_test.rb

```diff
@@ -0,0 +1,56 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  def setup
+    @user = User.new(name: 'Example User', email:'user@example.com')
+  end
+
+  test 'should be valid' do
+    assert @user.valid?
+  end
+
+  test 'name should be present' do
+    @user.name = ''
+    assert_not @user.valid?
+  end
+
+  test 'email should be present' do
+    @user.email = ''
+    assert_not @user.valid?
+  end
+
+  test 'name should not be too long' do
+    @user.name = 'a' * 51
+    assert_not @user.valid?
+  end
+
+  test 'email should not be too long' do
+    @user.email = 'a' * 244 + '@example.com'
+    assert_not @user.valid?
+  end
+
+  test 'email validation should accept valid addresses' do
+    valid_addresses = %w[user@example.com USER@foo.COM A_US-ER@foo.bar.org
```

Comment:
__[rubocop]__
* `%w`-literals should be delimited by `(` and `)`. [:ref](https://github.com/bbatsov/ruby-style-guide#percent-literal-braces)